bust-a-move-framework/src/main/java/bustamove/game/Cannon.java

```diff
@@ -190,73 +206,30 @@ public final void update(final GameContainer container, final int delta,
             final GameData gamehead) {
         timeShotFired += delta;
         game = gamehead;
-        if (this.playernr == 1) {
-            if (container.getInput().isKeyDown(Input.KEY_RIGHT)) {
-                if (container.getInput().isKeyPressed(Input.KEY_RIGHT)) {
-                    Log.log(this, "Cannon moving to the right");
-                }
-                this.timePassedKeyRight += delta;
-                if (this.timePassedKeyRight > INPUT_SCAN_DELAY) {
-                    stepDown();
-                    this.timePassedKeyRight = 0;
-                }
-            } else {
-                this.timePassedKeyRight = 0;
-            }
-            if (container.getInput().isKeyDown(Input.KEY_LEFT)) {
-                if (container.getInput().isKeyPressed(Input.KEY_LEFT)) {
-                    Log.log(this, "Cannon moving to the left");
+        for (int i = 0; i < 2; i++) {
+            if (container.getInput().isKeyDown(INPUTKEY[playernr - 1][i])) {
+                this.timePassedKey[i] += delta;

```

Comment:
Something I noticed. In 2 player mode: if one of the player presses the fire button, the timer is reset for both player. The both cannons will not auto shoot if at least pressed the fire button.